core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java

```diff
@@ -650,6 +650,17 @@ default void checkCanGrantExecuteFunctionPrivilege(SystemSecurityContext context
         denyGrantExecuteFunctionPrivilege(functionName, context.getIdentity(), granteeAsString);
     }
 
+    /**
+     * Check if identity is allowed to grant an access to the function execution to grantee.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    default void checkCanGrantExecuteFunctionPrivilege(SystemSecurityContext context, FunctionKind functionKind, CatalogSchemaRoutineName functionName, TrinoPrincipal grantee, boolean grantOption)
```

Comment:
I think we need the same for connector access control. Previous functions were not catalog related.

Reply:
To me that sounds more related to https://github.com/trinodb/trino/pull/12588 than to PTFs by themselves.
@kokosing i'd suggest this to be a follow-up.
@dain are you planning on taking care of this?